libsolidity/parsing/Parser.cpp

```diff
@@ -1183,11 +1183,21 @@ ASTPointer<Mapping> Parser::parseMapping()
 	}
 	else
 		fatalParserError(1005_error, "Expected elementary type name or identifier for mapping key type");
+	ASTPointer<ASTString> keyName;
+	if(m_scanner->currentToken() == Token::Identifier)
+		keyName = getLiteralAndAdvance();
+	else
+		keyName = make_shared<ASTString>("");
```

Comment:
Can you add a test for something like `mapping (uint uint => uint)`. Should be a parser error.